tests/dependencies.yml

```diff
@@ -0,0 +1,28 @@
+---
+- hosts: localhost
+  connection: local
+  remote_user: root
+  tasks:
+    - name: installing repo for Java 8 in Ubuntu
+      apt_repository: repo='ppa:openjdk-r/ppa'
+      when:
+        - ansible_os_family == 'Debian'
+        - ansible_distribution_version|version_compare(16.04, '<')
+
+- hosts: localhost
+  connection: local
+  remote_user: root
+  roles:
+    - role: geerlingguy.java
+      java_packages:
+        - openjdk-8-jdk
+      when: ansible_os_family == 'Debian'
+
+    - role: geerlingguy.java
+      java_packages:
+        - java-1.8.0-openjdk
+      when: ansible_os_family == 'RedHat'
```

Comment:
fully agree 👍  
Already >1 year ago I actually proposed the same (removing the java-dep) from another public role (ansible-spark), that we improved+use ourselves
and thought about this beeing a general problem in ansible dependency mgmt, resulting in https://github.com/ansible/ansible/issues/13215 ...